frontend/src/app/modules/availability/new-slot/schedule-definition/exclusion-date/exclusion-date.component.ts

```diff
@@ -1,24 +1,117 @@
+        super();
+    }
+
     @Input() set schedule(value: ISchedule | undefined) {
         this.scheduleValue = value ?? getDefaultSchedule(false);
+        this.scheduleValue.exclusionDates.forEach((date) =>
+            date.dayTimeRanges.forEach((range) => {
+                range.start = this.formatTime(range.start);
+                range.end = this.formatTime(range.end);
+            }));
+        this.updateExclusionDatesDisplay();
     }
 
-    formatTime = (time: string) => time.substring(0, 5);
+    updateExclusionDatesDisplay() {
+        this.exclusionDatesDisplay = getUpdatedExclusionDatesDisplay(
+            this.scheduleValue.exclusionDates,
+            this.scheduleValue.timeZone.timeValue,
+            this.dateAdapter,
+        );
+    }
 
     deleteExclusionDate(index: number) {
+        const exclusionDateDisplayToDelete = this.exclusionDatesDisplay[index];
+        const exclusionDateDisplaysToDeleteUtc = convertExclusionDateToOffset(
+            exclusionDateDisplayToDelete,
+            changeOffsetSign(this.scheduleValue.timeZone.timeValue),
+            this.dateAdapter,
+        );
+
+        exclusionDateDisplaysToDeleteUtc.forEach((date) => {
+            const sameDate = FindSameExclusionDateHelper(this.scheduleValue.exclusionDates, date);
+
+            if (sameDate) {
+                sameDate.dayTimeRanges = sameDate.dayTimeRanges.filter(
+                    (range) =>
+                        !date.dayTimeRanges.some(
+                            (rangeToDelete) => rangeToDelete.start === range.start && rangeToDelete.end === range.end,
+                        ),
+                );
+
+                sameDate.dayTimeRanges.forEach((range) => {
+                    date.dayTimeRanges.forEach((rangeToDelete) => {
+                        if (rangeToDelete.start === range.start && rangeToDelete.end < range.end) {
+                            range.start = rangeToDelete.end;
+                        } else if (rangeToDelete.end === range.end && rangeToDelete.start > range.start) {
+                            range.end = rangeToDelete.start;
+                        }
+                    });
+                });
+            }
+        });
+
         this.scheduleValue.exclusionDates = this.scheduleValue.exclusionDates.filter(
-            (date, dateIndex) => dateIndex !== index,
+            (date) => date.dayTimeRanges.length,
         );
+        this.updateExclusionDatesDisplay();
     }
+
+    showExclusionDatesWindow() {
+        const dialogConfig = new MatDialogConfig();
+
+        dialogConfig.data = this.scheduleValue.timeZone;
+        this.dialog
+            .open<ExclusionDatesPickerComponent, ITimeZone, IExclusionDate[] | undefined>(ExclusionDatesPickerComponent, dialogConfig)
+            .afterClosed()
+            .subscribe((newExclusionDates) => {
+                if (newExclusionDates) {
+                    newExclusionDates.forEach((newExclusionDate) => {
```

Comment:
why forEach?

Reply:
After user selects exclusion date with time ranges, according to selected timezone it can be converted to two exclusion dates in utc

Reply:
Fixed